### PR TITLE
Preact: Page Panel

### PR DESCRIPTION
**Pull request by aviettran**

### CODE REVIEW

File: src/client-main.ts [hunk 1]
```diff
@@ -786,7 +786,6 @@ const PS = new class extends PSModel {
 					}
 				}
 				this.update();
-				continue;
```
Comment (Zarel) [inline src/client-main.ts:789]: ?

Comment (aviettran) [reply src/client-main.ts]: Don't remember why I removed this. I thought I had a good reason, but I don't think I did.

File: src/panel-page.tsx [hunk 2]
```diff
@@ -52,15 +68,64 @@ function PagerLadderHelp(props: { room: PageRoom }) {
...
+				room.setHTMLData(args[2]);
+			}
+			return true;
+		case 'pagehtml':
+			room.setHTMLData(args[1]);
+			return true;
+		}
+	}
 	render() {
 		const { room } = this.props;
```
Comment (Zarel) [inline src/panel-page.tsx:123]:
```suggestion
		const {room} = this.props;
```

Comment (Zarel) [reply src/panel-page.tsx:123]: We don't have tslint but you know the rules and so do I!

File: src/panel-page.tsx [hunk 3]
```diff
@@ -52,15 +68,64 @@ function PagerLadderHelp(props: { room: PageRoom }) {
...
 		const RenderPage = () => {
```
Comment (Zarel) [inline src/panel-page.tsx:124]: You definitely seem to define components like `RenderPage` and `PageServerHTML` that I would just have let be functions. Like I would just use a variable for `RenderPage`, and `PageServerHTML` I would probably just implement inline.

File: src/panel-page.tsx [hunk 4]
```diff
@@ -8,17 +8,33 @@
...
+			PS.send(`|/join ${this.id}`);
+			this.connected = true;
+			this.connectWhenLoggedIn = false;
+		}
+	}
 }
 
-function PagerLadderHelp(props: { room: PageRoom }) {
+function PageLadderHelp(props: { room: PageRoom }) {
 	const { room } = props;
```
Comment (Zarel) [inline src/panel-page.tsx:38]:
```suggestion
	const {room} = props;
```

File: src/panel-page.tsx [hunk 5]
```diff
@@ -8,17 +8,33 @@
...
+		if (!this.connected) {
+			PS.send(`|/join ${this.id}`);
+			this.connected = true;
+			this.connectWhenLoggedIn = false;
+		}
+	}
 }
 
-function PagerLadderHelp(props: { room: PageRoom }) {
+function PageLadderHelp(props: { room: PageRoom }) {
```
Comment (Zarel) [inline src/panel-page.tsx:37]:
```suggestion
function PageLadderHelp(props: {room: PageRoom}) {
```

File: src/panel-page.tsx [hunk 6]
```diff
@@ -52,15 +68,64 @@ function PagerLadderHelp(props: { room: PageRoom }) {
...
+}
+
 class PagePanel extends PSRoomPanel<PageRoom> {
+	clientRooms: { [key: string]: JSX.Element } = { 'ladderhelp': <PageLadderHelp room={this.props.room}/> };
+
+	/**
+	 * @return true to prevent line from being sent to server
+	 */
+	receiveLine(args: Args) {
+		const { room } = this.props;
```
Comment (Zarel) [inline src/panel-page.tsx:89]:
```suggestion
		const {room} = this.props;
```

File: src/panel-page.tsx (hunk 2 above)
Comment (aviettran) [reply src/panel-page.tsx]: I have made these changes, but what are the rules exactly?

File: src/panel-page.tsx (hunk 3 above)
Comment (aviettran) [reply src/panel-page.tsx]: Revised.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**